From wfp5p@virginia.edu  Wed Jul 29 14:23:26 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Wed, 29 Jul 2009 17:00:10 -0400
Subject: Staging: hv: remove WaitEventClose()
To: greg@kroah.com
Cc: hjanssen@microsoft.com
Message-ID: <1248901215-29685-3-git-send-email-wfp5p@virginia.edu>


All WaitEventClose() close did was call kfree(), so get rid of it and
replace it with a call to kfree()

Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/hv/Channel.c     |    6 +++---
 drivers/staging/hv/ChannelMgmt.c |    2 +-
 drivers/staging/hv/Connection.c  |    4 ++--
 drivers/staging/hv/NetVsc.c      |    4 ++--
 drivers/staging/hv/RndisFilter.c |    2 +-
 drivers/staging/hv/StorVsc.c     |    4 ++--
 drivers/staging/hv/include/osd.h |    1 -
 drivers/staging/hv/osd.c         |    5 -----
 8 files changed, 11 insertions(+), 17 deletions(-)

--- a/drivers/staging/hv/Channel.c
+++ b/drivers/staging/hv/Channel.c
@@ -309,7 +309,7 @@ Cleanup:
 	REMOVE_ENTRY_LIST(&openInfo->MsgListEntry);
 	spin_unlock_irqrestore(&gVmbusConnection.channelmsg_lock, flags);
 
-	WaitEventClose(openInfo->WaitEvent);
+	kfree(openInfo->WaitEvent);
 	kfree(openInfo);
 
 	DPRINT_EXIT(VMBUS);
@@ -596,7 +596,7 @@ Cleanup:
 	REMOVE_ENTRY_LIST(&msgInfo->MsgListEntry);
 	spin_unlock_irqrestore(&gVmbusConnection.channelmsg_lock, flags);
 
-	WaitEventClose(msgInfo->WaitEvent);
+	kfree(msgInfo->WaitEvent);
 	kfree(msgInfo);
 
 	DPRINT_EXIT(VMBUS);
@@ -658,7 +658,7 @@ VmbusChannelTeardownGpadl(
 	REMOVE_ENTRY_LIST(&info->MsgListEntry);
 	spin_unlock_irqrestore(&gVmbusConnection.channelmsg_lock, flags);
 
-	WaitEventClose(info->WaitEvent);
+	kfree(info->WaitEvent);
 	kfree(info);
 
 	DPRINT_EXIT(VMBUS);
--- a/drivers/staging/hv/ChannelMgmt.c
+++ b/drivers/staging/hv/ChannelMgmt.c
@@ -762,7 +762,7 @@ VmbusChannelRequestOffers(
 Cleanup:
 	if (msgInfo)
 	{
-		WaitEventClose(msgInfo->WaitEvent);
+		kfree(msgInfo->WaitEvent);
 		kfree(msgInfo);
 	}
 
--- a/drivers/staging/hv/Connection.c
+++ b/drivers/staging/hv/Connection.c
@@ -155,7 +155,7 @@ VmbusConnect(void)
 	}
 
 
-	WaitEventClose(msgInfo->WaitEvent);
+	kfree(msgInfo->WaitEvent);
 	kfree(msgInfo);
 	DPRINT_EXIT(VMBUS);
 
@@ -183,7 +183,7 @@ Cleanup:
 	if (msgInfo)
 	{
 		if (msgInfo->WaitEvent)
-			WaitEventClose(msgInfo->WaitEvent);
+			kfree(msgInfo->WaitEvent);
 
 		kfree(msgInfo);
 	}
--- a/drivers/staging/hv/include/osd.h
+++ b/drivers/staging/hv/include/osd.h
@@ -134,7 +134,6 @@ extern int TimerStop(struct osd_timer *t
 extern void TimerStart(struct osd_timer *t, u32 expirationInUs);
 
 extern struct osd_waitevent *WaitEventCreate(void);
-extern void WaitEventClose(struct osd_waitevent *waitEvent);
 extern void WaitEventSet(struct osd_waitevent *waitEvent);
 extern int	WaitEventWait(struct osd_waitevent *waitEvent);
 
--- a/drivers/staging/hv/NetVsc.c
+++ b/drivers/staging/hv/NetVsc.c
@@ -880,7 +880,7 @@ Cleanup:
 
 	if (netDevice)
 	{
-		WaitEventClose(netDevice->ChannelInitEvent);
+		kfree(netDevice->ChannelInitEvent);
 
 		while (!IsListEmpty(&netDevice->ReceivePacketList))
 		{
@@ -963,7 +963,7 @@ NetVscOnDeviceRemove(
 		kfree(netvscPacket);
 	}
 
-	WaitEventClose(netDevice->ChannelInitEvent);
+	kfree(netDevice->ChannelInitEvent);
 	FreeNetDevice(netDevice);
 
 	DPRINT_EXIT(NETVSC);
--- a/drivers/staging/hv/osd.c
+++ b/drivers/staging/hv/osd.c
@@ -216,11 +216,6 @@ struct osd_waitevent *WaitEventCreate(vo
 	return wait;
 }
 
-void WaitEventClose(struct osd_waitevent *waitEvent)
-{
-	kfree(waitEvent);
-}
-
 void WaitEventSet(struct osd_waitevent *waitEvent)
 {
 	waitEvent->condition = 1;
--- a/drivers/staging/hv/RndisFilter.c
+++ b/drivers/staging/hv/RndisFilter.c
@@ -273,7 +273,7 @@ static inline void PutRndisRequest(RNDIS
 	REMOVE_ENTRY_LIST(&Request->ListEntry);
 	spin_unlock_irqrestore(&Device->request_lock, flags);
 
-	WaitEventClose(Request->WaitEvent);
+	kfree(Request->WaitEvent);
 	kfree(Request);
 }
 
--- a/drivers/staging/hv/StorVsc.c
+++ b/drivers/staging/hv/StorVsc.c
@@ -516,7 +516,7 @@ static int StorVscChannelInit(struct hv_
 Cleanup:
 	if (request->WaitEvent)
 	{
-		WaitEventClose(request->WaitEvent);
+		kfree(request->WaitEvent);
 		request->WaitEvent = NULL;
 	}
 
@@ -678,7 +678,7 @@ StorVscOnHostReset(
 	/* FIXME: Add a timeout */
 	WaitEventWait(request->WaitEvent);
 
-	WaitEventClose(request->WaitEvent);
+	kfree(request->WaitEvent);
 	DPRINT_INFO(STORVSC, "host adapter reset completed");
 
 	/*
